lib/declare_schema/model.rb

```diff
@@ -126,13 +114,35 @@ def belongs_to(name, scope = nil, **options)
           ActiveSupport::Deprecation.warn("belongs_to limit: is deprecated since it is now inferred")
         end
 
-        index_options = {}
-        index_options[:name]   = options.delete(:index) if options.has_key?(:index)
-        index_options[:unique] = options.delete(:unique) if options.has_key?(:unique)
-        index_options[:allow_equivalent] = options.delete(:allow_equivalent) if options.has_key?(:allow_equivalent)
+        # index: true means create an index on the foreign key
+        # index: false means do not create an index on the foreign key
+        # index: { ... } means create an index on the foreign key with the given options
+        index_value = options.delete(:index)
+        if index_value != false || options.has_key?(:unique) || options.has_key?(:allow_equivalent)
```

Comment:
`if index_value != false` looks strange, but is this to because we want to enter the block if `index_value` is nil?

Reply:
Yeah, that is really confusing. It might not be right still.

If it's `false`, we can still enter the `if` for other reasons. But then we'll `raise` to indicate that `false` is contradictory.

But if it's `nil`, you're right that that will enter the `if` even if nothing else is set. And the `case` will do nothing. None of the code will do anything except set `index_options = {}`. Which is then treated as truthy below... which seems like it would then call `index`. Hmm....

Reply:
Aha, I think I see how this is subtly broken. The base branch is slightly different, because it does:
```ruby
index_options[:name]   = options.delete(:index) if options.has_key?(:index)
```
So if `:index` isn't in options at all, it won't be copied across into `index_options`.

I think I need to check `has_key?` rather than `!= false`. And then clear it out with `delete` inside the `if`.

Reply:
Update: I put the code back closer to what it was, without any raise on contradiction. I put it back to where`index: false` takes precedence over any other sub-settings for indexes.